corehq/apps/custom_data_fields/edit_entity.py

```diff
@@ -137,13 +138,35 @@ def fields(self):
 
     def init_form(self, post_dict=None):
         fields = OrderedDict()
-        if domain_has_privilege(self.domain, privileges.APP_USER_PROFILES):
-            profiles = self.model.get_profiles()
+
+        from corehq.apps.users.views.mobile import UserFieldsView
+        has_profile_privilege_and_is_user_fields_view = (
+            domain_has_privilege(self.domain, privileges.APP_USER_PROFILES)
+            and self.field_view is UserFieldsView
+        )
+        if has_profile_privilege_and_is_user_fields_view:
+            original_profile_id = None
+            if self.existing_custom_data:
+                original_profile_id = self.existing_custom_data.get(PROFILE_SLUG, None)
+
+            profiles, can_edit_original_profile = self.field_view.get_displayable_profiles_and_edit_permission(
+                original_profile_id, self.domain, self.request_user
+            )
+
+            def validate_profile_slug(value):
+                from django.core.exceptions import ValidationError
+                if value not in [p.id for p in profiles]:
+                    raise ValidationError(
+                        _('Invalid profile selected. Please select a valid profile.'),
+                    )
```

Comment:
yes. When the the field is disabled, `profiles` (retrieved from `get_displayable_profiles_and_edit_permission`) will contain the profile that is already set for the user (which is also be the profile that `value` references)

Reply:
no it won't prevent them from editing other data fields. The `profiles` list comes from `get_displayable_profiles_and_edit_permission`.

 If the user is not permitted to edit the original profile, it returns a list containing the original profile. ` profiles ={CustomDataFieldsProfile.objects.get(id=original_profile_id)}`. So the validation will pass since it'll be validating that the original profile is in that list (Which it will be).